admin/testbrew.py

```diff
@@ -0,0 +1,112 @@
+        * A snapshot with a particular name, homebrew-clean
+        * Homebrew installed and available
+"""
+
+import os
+import sys
+import urllib2
+
+from twisted.python.usage import Options, UsageError
+
+from subprocess import check_output, CalledProcessError
+
+from flocker.provision._install import run_with_fabric, task_test_homebrew
+
+
+YOSEMITE_VMX_PATH = (
+    '"'
+    "{home}/Documents/Virtual Machines.localized/"
+    "OS X 10.10.vmwarevm/OS X 10.10.vmx"
+    '"'
+).format(home=os.getenv("HOME"))
+
+YOSEMITE_SNAPSHOT = "homebrew-clean"
+
+VM_USERNAME = "ClusterHQVM"
+
+VM_HOST = "10.0.126.88"
+
+
+class TestBrewOptions(Options):
+    description = ("Start a VMWare OS X VM and test a brew recipe "
+                   "installation.")
+
+    optParameters = [
+        ['vmhost', 'h', VM_HOST,
+         'IP address or hostname of the Virtual Machine'],
+        ['vmuser', 'u', VM_USERNAME,
+         'Username for the Virtual Machine'],
+        ['vmpath', 'p', YOSEMITE_VMX_PATH,
+         'Full path to the Virtual Machine image'],
+        ['vmsnapshot', 's', YOSEMITE_SNAPSHOT,
+         'Snapshot identifier for the Virtual Machine'],
+    ]
+
+    synopsis = ('Usage: test-brew-recipe <recipe URL> [--vmhost <host>] '
+                '[--vmuser <username>] '
+                '[--vmpath <path>] '
+                '[--vmsnapshot <name>]')
+
+    def parseArgs(self, *args):
+        if len(args) < 1:
+            raise UsageError((
+                "URL of homebrew recipe not specified. "
+                "Run with --help for usage"
+            ))
+        else:
+            self['recipe_url'] = args[0]
+
+
+def main(args):
+    try:
+        options = TestBrewOptions()
+        try:
+            options.parseOptions(args)
+        except UsageError as e:
+            sys.stderr.write("Error: {error}.\n".format(error=str(e)))
+            sys.exit(1)
+        recipe_url = options['recipe_url']
+        # Open the recipe URL just to validate and verify that it exists.
+        # We do not need to read its content.
+        urllib2.urlopen(recipe_url)
+        check_output([
+            "vmrun", "revertToSnapshot",
+            options['vmpath'], options['vmsnapshot'],
+        ])
+        check_output([
+            "vmrun", "start", options['vmpath'], "nogui",
+        ])
+        commands = task_test_homebrew(recipe_url)
+        run_with_fabric(options['vmuser'], options['vmhost'],
+                        commands=commands)
+        check_output([
+            "vmrun", "stop", options['vmpath'], "hard",
+        ])
+        print "Done."
+        sys.exit(0)
+    except CalledProcessError as e:
+        sys.stderr.write(
+            (
+                "Error: Command {cmd} terminated with exit status {code}.\n"
+            ).format(cmd=" ".join(e.cmd), code=e.returncode)
+        )
+        sys.exit(1)
+    except Exception as e:
+        sys.stderr.write("Error: {error}.\n".format(error=str(e)))
+        sys.exit(1)
```

Comment:
is there a particular exception you wanted this block to handle or is it really the catch-all fallback case it seems to be?  Since this is a developer-facing tool I think it's probably better to let the unexpected exception reach the top-level and get a traceback reported.  It's going to be hard to debug if this handler is left in because only the exception message will be visible - no context.  Throwing tracebacks at end-users isn't nice because they'll get confused but throwing them at developers is nice because then they have more information about how to fix the bug.